pkg/registry/core/service/allocator/bitmap.go

```diff
@@ -70,6 +71,35 @@ func NewAllocationMap(max int, rangeSpec string) *AllocationBitmap {
 		rangeSpec: rangeSpec,
 	}
 	return &a
+
+}
+
+// defaultReservedValues defines the number of values reserved for statically allocation.
+const defaultReservedValues = 32
+
+// NewAllocationReservedMap creates an allocation bitmap using a random scan strategy that,
+// if the maximum value is greater than 64, is subdivided in two blocks.
+// The first block of values will not be used for random value assigned by the AllocateNext()
+// method until the second block of values has been exhausted.
+func NewAllocationReservedMap(max int, rangeSpec string) *AllocationBitmap {
```

Comment:
We need to create a new constructor because `NewAllocationMap` is used for the NodePort allocator too